core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildContainerRunner.java

```diff
@@ -39,7 +39,8 @@ public NativeImageBuildContainerRunner(NativeConfig nativeConfig, Path outputDir
 
         outputPath = outputDir == null ? null : outputDir.toAbsolutePath().toString();
         containerName = "build-native-" + RandomStringUtils.random(5, true, false);
-        this.needJava17Image = javaVersion.isExactlyJava11() == CompiledJavaVersionBuildItem.JavaVersion.Status.FALSE;
+        // Default to using java17 builder images
+        this.needJava17Image = true;
```

Comment:
should `NativeConfig#getEffectiveBuilderImage` also be adjusted?

Reply:
We can either go full on and adjust `NativeConfig#getEffectiveBuilderImage` and remove `needJava17Image` as well or we can go with this change "for now" since `NativeConfig#getEffectiveBuilderImage` is only invoked twice in our code base, both times using `needJava17Image` as a parameter.

I am OK with both approaches :)

Reply:
No trong objections either - gradual is perhaps better, but it seems `UpxCompressionBuilder` also depends on it and we probably want that to behave consistently.

Reply:
OK, I somehow missed that reference which is a good indicator why it's probably better to go full on :)

Please re-review